backend/src/layer/data/flickr_api/api_request.rs

```diff
@@ -0,0 +1,215 @@
+        debug!("request_license finished: {:?}", res);
+        println!("PRINT VALUE: {:?}", res); //TODO remove
+        Ok(res)
+    }
+
+    /// Sends a url request to the FlickrApi.
+    /// # Return
+    /// The [`JsonRootError`] struct, containing the api response.
+    /// # Errors
+    /// If the request could not be decoded to json or the connection could not be established, an error will be returned.
+    async fn request_err(&self, url: &String) -> Result<JsonRootError, ImageHosterError> {
+        let res = reqwest::get(url).await.map_err(|_| ImageHosterError::NotConnected)?.json::<JsonRootError>().await.map_err(|_| ImageHosterError::DecodeFailed)?;
+        debug!("request_err finished: {:?}", res);
+        println!("PRINT VALUE: {:?}", res);
+        Ok(res)
+    }
+
+    /// This method creates an api request url.
+    /// This url is used to create an request with the [`request_sizes`] function.
+    /// # Errors
+    /// If the request could not be decoded ([`ImageHosterError::DecodeFailed`],
+    /// another request with [`request_err`] will be attempted.
+    /// This error request returns a more detailed error information.
+    /// # Returns
+    /// An Error (as above mentioned) or an [`ImageMetaData`] struct containing information about the requested image.
+    pub async fn flickr_photos_get_sizes(
+        &self,
+        photo_id: &str,
+    ) -> Result<ImageMetaData, ImageHosterError> {
+        let url = &format!(
+            "{BASE_URL}{GET_SIZES}{TAG_API_KEY}{api_key}{TAG_PHOTO_ID}{photo_id}{JSON}",
+            BASE_URL = Self::BASE_URL,
+            GET_SIZES = Self::GET_SIZES,
+            TAG_API_KEY = Self::TAG_API_KEY,
+            api_key = self.api_key,
+            TAG_PHOTO_ID = Self::TAG_PHOTO_ID,
+            JSON = Self::FORMAT
+        );
+        println!("flickr.photos.getSizes URL: {}", url); // TODO remove
+        match self.request_sizes(&url).await {
+            Ok(sizes) => Ok(self.parser.parse_get_sizes(sizes, photo_id)),
+            Err(e) => Err(self.determine_error(url, e).await)
+        }
+    }
+
+    /// This method creates an api request url.
+    /// This url is used to create an request with the [`request_license`] function.
+    /// # Errors
+    /// If the request could not be decoded ([`ImageHosterError::DecodeFailed`],
+    /// another request with [`request_err`] will be attempted.
+    /// This error request returns a more detailed error information.
+    /// # Returns
+    /// An Error (as above mentioned).
+    /// True if the given image is hosted under a valid license (see [`json_parser::get_valid_licences`] for more info).
+    /// False if not.
+    pub async fn flickr_photos_licenses_get_license_history(
+        &self,
+        photo_id: &str,
+    ) -> Result<bool, ImageHosterError> {
+        let url = &format!(
+            "{BASE_URL}{GET_LICENCE_HISTORY}{TAG_API_KEY}{api_key}{TAG_PHOTO_ID}{photo_id}{JSON}",
```

Comment:
we call them images (not potos); but if that is how flickr calls them its probably ok